aiohttp/client.py

```diff
@@ -150,6 +156,34 @@
     SSLContext = None
 
 
+class _RequestOptions(TypedDict, total=False):
+    params: Union[Mapping[str, str], None]
+    data: Any
+    json: Any
+    cookies: Union[LooseCookies, None]
+    headers: Union[LooseHeaders, None]
+    skip_auto_headers: Union[Iterable[str], None]
+    auth: Union[BasicAuth, None]
+    allow_redirects: bool
```

Comment:
The Unpack does have the downside of losing information about defaults. However, I'm not aware of a solution that allows documenting defaults (without duplicating everything)...

Reply:
That's mainly why I left the other methods alone.

Reply:
If you mean ._request(), that doesn't really make a difference as it's not exposed to the user. For anything the user looks at, we've lost the defaults.